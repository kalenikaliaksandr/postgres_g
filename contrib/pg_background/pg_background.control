comment = 'Run SQL queries in the background'
default_version = '1.0'
module_pathname = '$libdir/pg_background'
relocatable = true
