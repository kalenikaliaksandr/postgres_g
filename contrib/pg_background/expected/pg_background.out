CREATE EXTENSION pg_background;
--run 5 workers which wait about 1 second
CREATE TABLE input as SELECT x FROM generate_series(1,5,1) x ORDER BY x DESC;
CREATE TABLE output(place int,value int);
CREATE sequence s start 1;
CREATE TABLE handles as SELECT pg_background_launch('select pg_sleep('||x||'); insert into output values (nextval(''s''),'||x||');') h FROM input;
SELECT (SELECT * FROM pg_background_result(h) as (x text) limit 1) FROM handles;
    x     
----------
 SELECT 1
 SELECT 1
 SELECT 1
 SELECT 1
 SELECT 1
(5 rows)

SELECT * FROM output ORDER BY place;
 place | value 
-------+-------
     1 |     1
     2 |     2
     3 |     3
     4 |     4
     5 |     5
(5 rows)

